tests/cpp_unit_tests/test_math.cpp

```diff
@@ -0,0 +1,364 @@
+    -1.25000000e-01, 4.37500000e-01, -2.89062500e-01, -8.98437500e-02};
+
+  int nums = n / 2 + 1;
+  std::vector<double> test_vals(nums, 0);
+  openmc::calc_zn_rad(n, rho, test_vals.data());
+
+  REQUIRE_THAT(ref_vals, Catch::Matchers::Approx(test_vals));
+}
+
+TEST_CASE("Test rotate_angle")
+{
+  std::vector<double> uvw0 {1.0, 0.0, 0.0};
+  double phi = 0.0;
+
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<> dist(0, 1);
+
+  uint64_t prn_seed = 0;
+  for (int i = 0; i < 63; ++i) {
+    prn_seed |= static_cast<std::uint64_t>(dist(gen)) << i;
+  }
+
+  // reference: mu of 0 pulls the vector the bottom, so:
+  SECTION("Test rotate_angle mu is 0")
+  {
+    std::vector<double> ref_uvw {0.0, 0.0, -1.0};
+
+    double mu = 0.0;
+
+    std::vector<double> test_uvw(uvw0);
+    openmc::rotate_angle_c(test_uvw.data(), mu, &phi, &prn_seed);
+
+    REQUIRE_THAT(ref_uvw, Catch::Matchers::Approx(test_uvw));
+  }
+
+  // Repeat for mu = 1 (no change)
+  SECTION("Test rotate_angle mu is 1")
+  {
+    std::vector<double> ref_uvw = {1.0, 0.0, 0.0};
+
+    double mu = 1.0;
+
+    std::vector<double> test_uvw(uvw0);
+    openmc::rotate_angle_c(test_uvw.data(), mu, &phi, &prn_seed);
+
+    REQUIRE_THAT(ref_uvw, Catch::Matchers::Approx(test_uvw));
+  }
+
+  // Now to test phi is None
+  SECTION("Test rotate_angle no phi")
+  {
+    // When seed = 1, phi will be sampled as 1.9116495709698769
+    // The resultant reference is from hand-calculations given the above
+    std::vector<double> ref_uvw = {
+      0.9, -0.422746750548505, 0.10623175090659095};
+
+    double mu = 0.9;
+    prn_seed = 1;
+
+    std::vector<double> test_uvw(uvw0);
+    openmc::rotate_angle_c(test_uvw.data(), mu, NULL, &prn_seed);
+
+    REQUIRE_THAT(ref_uvw, Catch::Matchers::Approx(test_uvw));
+  }
+}
+
+TEST_CASE("Test maxwell_spectrum")
+{
+  double ref_val = 0.27767406743161277;
+
+  double T = 0.5;
+  uint64_t prn_seed = 1;
+
+  double test_val = openmc::maxwell_spectrum(T, &prn_seed);
+
+  REQUIRE(ref_val == test_val);
+}
+
+TEST_CASE("Test watt_spectrum")
+{
+  double ref_val = 0.30957476387766697;
+
+  double a = 0.5;
+  double b = 0.75;
+  uint64_t prn_seed = 1;
+
+  double test_val = openmc::watt_spectrum(a, b, &prn_seed);
+
+  REQUIRE(ref_val == test_val);
+}
+
+TEST_CASE("Test normal_variate function")
+{
+
+  // Generate a series of normally distributed random numbers and test
+  // whether their mean and standard deviation are close to the expected value
+  SECTION("Test with non-zero standard deviation")
+  {
+    uint64_t seed = 1;
+
+    double mean = 0.0;
+    double standard_deviation = 1.0;
+
+    int num_samples = 10000;
+    double sum = 0.0;
+    double sum_squared_difference = 0.0;
+
+    for (int i = 0; i < num_samples; ++i) {
+      double sample = openmc::normal_variate(mean, standard_deviation, &seed);
+      sum += sample;
+      sum_squared_difference += (sample - mean) * (sample - mean);
+      seed += 1;
```

Comment:
I'm pretty sure you don't need to touch the seed here, since `normal_variate` will advance it.

Reply:
Got it, thank you so much for your review